File: programmers/Level2/위장.py
#해시
def solution(clothes):
    clothes_table = {}
    count = 1
    
    for i in clothes:
        #입은 의상에 같은 종류가 있으면
        if i[1] in clothes_table.keys():
            clothes_table[str(i[1])] += 1
        else:
            clothes_table[str(i[1])] = 1

    for i in clothes_table.values():
        count *= (i+1)
    
    answer = count - 1
    return answer

File: JAVA_programmers/Level 2/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String, ArrayList<String>> cMap = new HashMap<>();
        for(String[] c: clothes){
            if(!cMap.containsKey(c[1])) cMap.put(c[1], new ArrayList<>());
            cMap.get(c[1]).add(c[0]);
        }
        
        for(Map.Entry<String, ArrayList<String>> e:cMap.entrySet()){
            answer *= (e.getValue().size()+1);
        }
        return answer-1;
    }
}
